tests/Tests/Unit/Transport/RestTransportTest.php

```diff
@@ -144,6 +146,91 @@ public function testStartUnaryCallThrowsException()
             ->wait();
     }
 
+    public function testStartUnaryCallWithValidProtoNotLoadedInDescPool()
+    {
+        $endpoint = 'www.example.com';
+        $expectedRequest = new Request(
+            'POST',
+            $endpoint,
+            [],
+            ''
+        );
+        $body = [
+            'name' => 'projects/my-project/locations/us-central1/operations/my-operation',
+            'metadata' => [
+                // This type is arbitrarily chosen and should not exist within the descriptor pool
+                // upon instantation of this test.
+                '@type' => 'type.googleapis.com/google.type.DateTime'
+            ]
+        ];
+        $httpHandler = function (RequestInterface $request) use ($body, $expectedRequest) {
+            $this->assertEquals($expectedRequest, $request);
+            return Create::promiseFor(
+                new Response(
+                    200,
+                    [],
+                    json_encode($body)
+                )
+            );
+        };
+        $call = new Call(
+            'Testing123',
+            Operation::class,
+            new MockRequest()
+        );
+
+        $response = $this->getTransport($httpHandler, $endpoint)
+            ->startUnaryCall($call, [
+                'metadataReturnType' => DateTime::class
+            ])
+            ->wait();
+
+        $this->assertInstanceOf(Operation::class, $response);
+        $this->assertEquals(
+            $body['metadata']['@type'],
+            $response->getMetadata()->getTypeUrl()
+        );
+    }
```

Comment:
It's unlikely, but I'm a little worried that this type could be accidentally loaded earlier. Could we mitigate this by adding a `@runInSeparateProcess` PHPUnit annotation?

Reply:
Good call, added!